File: app/src/main/java/com/rye/catcher/dbs/Person.java
package com.rye.catcher.dbs;

import io.realm.RealmList;
import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;
//realm
public class Person extends RealmObject {
    @PrimaryKey
    private long id;
    private boolean sex;
    private String name;
    private String home_address;
    private String job;
    //与狗关联
    private RealmList<Dog> dogs;

    public RealmList<Dog> getDogs() {
        return dogs;
    }

    public void setDogs(RealmList<Dog> dogs) {
        this.dogs = dogs;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public boolean isSex() {
        return sex;
    }

    public void setSex(boolean sex) {
        this.sex = sex;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getHome_address() {
        return home_address;
    }

    public void setHome_address(String home_address) {
        this.home_address = home_address;
    }

    public String getJob() {
        return job;
    }

    public void setJob(String job) {
        this.job = job;
    }
}

File: app/src/main/java/com/rye/catcher/dbs/Person.kt
package com.rye.catcher.dbs

import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import io.realm.annotations.Required

 open class Person:RealmObject() {
    @PrimaryKey
    var id : Long ?= 0
    var sex: Boolean ?= false
    @Required
    var name:String ?= null
    var home_address:String ?= null
    var job:String ?= null
    var company_address:String ?= null
    var interest:String ?=null
    var singleDog:Boolean ?=true
    var dogs:RealmList<Dog> ?= null

}